keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -395,7 +395,7 @@ class ClusterHandler(
         if (them.distinctBy { it.launchConfiguration.appVersion }.size == 1) {
           publisher.publishEvent(ArtifactVersionDeployed(
             resourceId = resource.id,
-            artifactVersion = them.first().launchConfiguration.appVersion
+            artifactVersion = them.first().launchConfiguration.appVersion!!
```

Comment:
Isn't '!!' throws 'KotlinNullPointerException'? Do we want it here?

Reply:
Not exactly. It converts a nullable type (say `String?`) to the non-nullable (`String`) counter-part, and only throws the KNPE if the expression is null. You generally only use when you're sure the value won't be null but the compiler can't infer it. I need to add a check here though to make sure it's not.